core/arns/src/test/java/software/amazon/awssdk/arns/ArnResourceTest.java

```diff
@@ -64,4 +66,72 @@ public void hashCodeEquals_minimalProperties() {
         assertThat(arnResource.equals(anotherResource)).isTrue();
         assertThat(arnResource.hashCode()).isEqualTo(anotherResource.hashCode());
     }
+
+    @Test
+    public void toString_resourceIdOnly() {
+        ArnResource arnResource = ArnResource.builder().resourceId("resource").build();
+        assertThat("null:resource:null").isEqualTo(arnResource.toString());
```

Comment:
Add unit test to demonstrate the unexpected 'null' behavior here of the toString